### PR TITLE
(Feat/dsm):Sentiment meter

### PR DESCRIPTION
**Pull request by kartik162000**
- Api endpoints for creating sentiment, getting all sentiments and by Id, getting the "percentage" of sentiment for a particular date.
- Swagger documentation for all end points.
- Validation using joi and middleware.
- Update swagger docs, controllers, services according to new design
- **Add delete endpoint for deleting a particular sentiment by sentimentId**
**GET request to display all sentiments**
image

**POST request to make new sentiment**
image

**GET request to get Percentage count for a particular date**
image

**GET request to get detail for a particular id**
image

**PATCH  to update sentiment**
image

**DELETE to delete a sentiment**
Screenshot 2023-03-13 at 8 29 48 PM

### CODE REVIEW

File: src/routes/api/index.js [hunk 1]
```diff
@@ -3,5 +3,6 @@ const router = express.Router();
 
 
 router.use('/po-notes', require('./poNotes.routes'));
+router.use('/dsm-sentiment-meter',require('./dsm/sentimentMeter.routes'))
```
Comment (karthik01455) [inline src/routes/api/index.js:7]: empty line can be removed

File: src/routes/api/index.js [hunk 2]
```diff
@@ -3,5 +3,6 @@ const router = express.Router();
```
Comment (karthik01455) [inline src/routes/api/index.js:3]: empty line can be removed

File: src/routes/api/index.js [hunk 3]
```diff
@@ -3,5 +3,6 @@ const router = express.Router();
 
 
 router.use('/po-notes', require('./poNotes.routes'));
+router.use('/dsm-sentiment-meter',require('./dsm/sentimentMeter.routes'))
```
Comment (Abhishek-Kharpal) [inline src/routes/api/index.js:6]: Can we change the route to /dsm/sentiment-meter?

Comment (kartik162000) [reply src/routes/api/index.js:6]: ok, will do it

**Review by Harshit-Soni-Mck — APPROVED** [review]

**Review by sugumar-mani — APPROVED** [review]

**Comment by samim-hossain-mondal** [comment]
Resolved all conflicts ! Added a new delete endpoint (delete by sentimentId), kindly approve it if all are good

**Review by karthik01455 — APPROVED** [review]
looks good to me and conflicts resolved !

**Pull request merged**